tests/test_google_storage.py

```diff
@@ -486,3 +560,37 @@ def test_delete_directory_deletes_blobs_with_prefix(self) -> None:
         mock_unversioned_blobs[0].delete.assert_called_once_with()
         mock_unversioned_blobs[1].delete.assert_called_once_with()
         mock_unversioned_blobs[2].delete.assert_called_once_with()
+
+    def test_delete_directory_raises_when_empty(self) -> None:
```

Comment:
This test is actually testing the case where the file was removed between the `list_blobs` call and the `delete` call, not when the directory is empty.